pysteps/io/importers.py

```diff
@@ -200,6 +200,26 @@ def _get_grib_projection(grib_msg):
     return projparams
 
 
+def _get_threshold_value(precip):
+    """
+    Get the the rain/no rain threshold with the same unit, transformation and
+    accutime of the data.
+    If all the values are NaNs, the returned value is `np.nan`.
+    Otherwise, np.min(precip[precip > precip.min()]) is returned.
+
+    Returns
+    -------
+
+    threshold : float
+    """
+    valid_mask = np.isfinite(precip)
+    if valid_mask.any():
+        _precip = precip[valid_mask]
+        return np.min(_precip[_precip > _precip.min()])
```

Comment:
Line 218 will fail if precip is a uniform field (e.g. all zeros).
I think we need to add an additional test to line 216, something like:
`if valid_mask.any() and (precip[valid_mask] > precip[valid_mask].min()).any():`
or
`if valid_mask.any() and not (precip[valid_mask] == precip[valid_mask][0]).all():`
you get the point :-)

Reply:
Commit tf4614c0 fixes that bug.